versioned_docs/version-V2/use-cases/private-voting.md

```diff
@@ -0,0 +1,96 @@
+2. Register ballot topics.
+3. Register voters.
+4. Record votes.
+
+### Prerequisites
+
+For this use case, assume you have deployed the following:
+
+- **NFT**: Proof of Attendance (POAP) or social token that your community can mint on Ethereum.
+- **Contract**: Smart contract based on Semaphore and deployed on Ethereum.
+- **Dapp**: Your application that provides a user-friendly interface to Ethereum and the Semaphore contract.
+
+### Verify members
+
+For this use case, members are users who own the NFT in their Ethereum wallet.
+To verify membership, your Dapp needs to check the user's wallet for the NFT.
+To grant access to the wallet, the user clicks a `Connect wallet` button in the Dapp.
+Once the user approves access, the Dapp checks for the NFT in the Ethereum account.
+
+:::info
+Learn how to use [Ethers.js]() to an [check Ethereum wallets]().
+:::
+
+The Dapp allows the member to do the following:
+
+- [Create a poll](#create-a-poll).
+- [Register to vote](#register-voters).
+- [Vote on a topic](#record-votes).
+
+### Create a poll
+
+A poll is a Semaphore **group** that stores the following:
+- A topic.
+- The public ID of the poll creator.
+- Semaphore IDs of members who joined the poll.
+
+To create a poll, a member clicks a `Submit topic` button in the Dapp that then calls a contract function--for example:
+
+```ts
+SemaphoreVoting.createPoll
+```
+
+The member enters a topic and signs the transaction to save the poll on Ethereum.
+Next, learn how to [register voters](#register-voters) for the poll. 
+
+:::info
+To prevent double-voting, Semaphore generates an **external nullifier** for the topic.
+Learn how Semaphore uses [nullifiers]() to prevent double-voting and double-spending.
+:::
+
+### Register voters
+
+A voter is a [member](#members) that joins a poll to vote.
+To join a poll, a member uses the Dapp for the following contract interactions:
+
+1. Member selects a poll (topic) to join.
+2. Dapp calls a contract function to create a private identity for the member.
+3. Wallet prompts the Member to sign the transaction and create the identity.
+4. Dapp passes the private identity to a contract function that creates a public anonymous Semaphore ID (identity commitment).
+5. Dapp calls a contract function that adds the new Semaphore ID to the poll.
+6. Wallet prompts the Member to sign the transaction and approve joining the poll.
+
+With the member registered for a poll, learn how the Dapp [records votes on a topic](#record-votes).
+
+### Record votes
+
+To vote on a topic, a member registers in a poll and votes on the poll topic.
+The member signs a vote with a private on-chain or off-chain identity that remains provable and anonymous.
+
+In the Dapp, when a member clicks to vote on a poll topic, the Dapp calls the contract to do the following:
+1. Create a proof that the member is registered for the poll and hasn't already voted in the poll. See how Semaphore [verifies proofs and prevents double-voting]().
```

Comment:
Step 1 happens off-chain, there is no interaction with contracts. `@zk-kit/protocols` is used to generate zk proofs off-chain.